samples/core/visualization/html.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import kfp.dsl as dsl
+import kfp.components as comp
+
+# Advanced function
+# Demonstrates imports, helper functions and multiple outputs
+from typing import NamedTuple
+
+def html_visualization(gcsPath: str) -> NamedTuple('VisualizationOutput', [('mlpipeline_ui_metadata', 'UI_metadata')]):
+    import json
+
+    metadata = {
+        'outputs': [{
+            'type': 'web-app',
+            'storage': 'inline',
+            'source': '<h1>Hello, World!</h1>',
+        }]
+    }
+
+    # Temporarily hack for empty string scenario: https://github.com/kubeflow/pipelines/issues/5830
+    if gcsPath and gcsPath != "" and gcsPath != "BEGIN-KFP-PARAM[]END-KFP-PARAM":
```

Comment:
`BEGIN-KFP-PARAM[]END-KFP-PARAM` seems like a new bug you discovered.
Consider adding some explanation in the existing issue?

Reply:
Also I think this can be simplified as `if gcsPath and gcsPath != "BEGIN-KFP-PARAM[]END-KFP-PARAM"` because "" is evaluated as False in Python.

Another nit may be considering using single quote for consistency

Reply:
Thank you both for the ideas! I have commented in the existing issue about this hack, and applied the fix regarding `if condition`.